examples/bri-3/src/bri/identity/bpiSubjects/persistence/bpiSubjects.repository.ts

```diff
@@ -34,4 +46,29 @@ export class BpiSubjectRepository extends PrismaService {
             newBpiSubjectModel.type, 
             newBpiSubjectModel.publicKey);
     }
+
+    async updateBpiSubject(bpiSubject: BpiSubject): Promise<BpiSubject> {
+        const newBpiSubjectModel = await this.bpiSubject.update({
+            where: { id: bpiSubject.id },
+            data: {
+                name: bpiSubject.name,
+                description: bpiSubject.description,
+                publicKey: bpiSubject.publicKey,
+                type: bpiSubject.type
+            }
+        });
+
+        return new BpiSubject(
+            newBpiSubjectModel.id, 
+            newBpiSubjectModel.name, 
+            newBpiSubjectModel.description, 
+            newBpiSubjectModel.type, 
+            newBpiSubjectModel.publicKey);
+    }
+
+    async deleteBpiSubject(bpiSubject: BpiSubject): Promise<void> {
+        await this.bpiSubject.delete({
```

Comment:
this is not returned and function is marked to have Promise<void> return type, or i am missing something

Reply:
the method is async. This is similar to Task return type in net core.

Reply:
ah ok, i was just using async in js context, makes sense